addons/stock/models/stock_move.py

```diff
@@ -178,12 +179,47 @@ def _default_group_id(self):
     next_serial = fields.Char('First SN')
     next_serial_count = fields.Integer('Number of SN')
     orderpoint_id = fields.Many2one('stock.warehouse.orderpoint', 'Original Reordering Rule', check_company=True)
+    availability_indication = fields.Char('Availability', compute='_compute_availability_indication')
+    availability_state = fields.Selection([
+        ('available', 'Available'),
+        ('waiting', 'Should be available on time'),
+        ('partial', 'Only a part can be reserved'),
+        ('late', 'Will not be available at time'),
+    ], compute='_compute_availability_indication')
 
     @api.onchange('product_id', 'picking_type_id')
     def onchange_product(self):
         if self.product_id:
             self.description_picking = self.product_id._get_description(self.picking_type_id)
 
+    @api.depends('date', 'date_expected', 'product_id.route_ids', 'product_type', 'product_uom_qty', 'reserved_availability')
+    def _compute_availability_indication(self):
+        mto_route = self.env.ref('stock.route_warehouse0_mto')
+        today = date.today()
+        for move in self:
+            if move.product_type != 'product' or move.reserved_availability == move.product_uom_qty:
+                move.availability_state = 'available'
+                precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+                move.availability_indication = float_repr(move.reserved_availability, precision)
+            elif not float_is_zero(move.reserved_availability, precision_rounding=move.product_id.uom_id.rounding):
+                move.availability_indication = move.reserved_availability
+                move.availability_state = 'partial'
+            elif mto_route in move.product_id.route_ids:
```

Comment:
that's not because you have this route set on the product that it'll be used, you could use move.procure_method == 'make_to_order'  but i'm not sure how it'll behave with a pick pack ship without buy. actually i don't know how this report should behave in this case